dkpro-core-io-bincas-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/bincas/BinaryCasReader.java

```diff
@@ -177,6 +253,14 @@ private TypeSystemImpl readTypeSystem() throws IOException {
             typeSystem = casMgr.getTypeSystem();
             typeSystem.commit();
         }
+
+		// If we encounter a Java-serialized file with an external
+		// TSI, then we reinitalize the CAS with the external TSI
+		// prior to loading the data
+		if (header[0] == (byte) 0xAC && header[1] == (byte) 0xED) {
```

Comment:
This method is called `readTypeSystem` and it has a return value. It is unexpected that it has a side-effect of reinitializing the CAS. I think it would be better to remove the `header` and `aCAS` parameters here and to introduce a new method to be called after `readTypeSystem`, maybe `initCasFromEmbeddedTS`?